src/Symfony/Component/JsonCrawler/JsonCrawler.php

```diff
@@ -0,0 +1,398 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonCrawler;
+
+use Symfony\Component\JsonCrawler\Exception\InvalidInputJsonException;
+use Symfony\Component\JsonCrawler\Exception\InvalidJsonPathException;
+use Symfony\Component\JsonCrawler\Exception\JsonCrawlerException;
+use Symfony\Component\JsonCrawler\Tokenizer\JsonPathToken;
+use Symfony\Component\JsonCrawler\Tokenizer\JsonPathTokenizer;
+use Symfony\Component\JsonCrawler\Tokenizer\TokenType;
+
+/**
+ * Crawls a JSON document using a JSON Path as described in the RFC 9535.
+ *
+ * @see https://datatracker.ietf.org/doc/html/rfc9535
+ *
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ *
+ * @experimental
+ */
+final class JsonCrawler implements JsonCrawlerInterface
+{
+    private const RFC9535_FUNCTIONS = [
+        'length' => true,
+        'count' => true,
+        'match' => true,
+        'search' => true,
+        'value' => true,
+    ];
+
+    private mixed $data;
+    private JsonPath $evaluatedPath;
+
+    public function __construct(string $json)
+    {
+        try {
+            $this->data = json_decode($json, true, 512, \JSON_THROW_ON_ERROR);
+        } catch (\JsonException $e) {
+            throw new InvalidInputJsonException($e->getMessage());
+        }
+    }
+
+    public function find(string|JsonPath $query): array
```

Comment:
It can be interesting to enforce return type to `array<array|string|float|int|bool|null>`

Reply:
Definitely! I added the relevant `@return`, thanks 👍